### PR TITLE
Async Serve Command

### PR DESCRIPTION
**Pull request by 0xTim**
Migrate `ServeCommand` to an `AsyncCommand` to enable proper custom executor support and remove any calls to `wait()`

### CODE REVIEW

**Comment by 0xTim** [comment]
Need to work out why the configuration has broken in the tests

**Comment by 0xTim** [comment]
Verified this works with https://github.com/vapor/template/pull/131/files with Leaf and Fluent and all working

File: Sources/XCTVapor/XCTApplication.swift [hunk 1]
```diff
@@ -89,41 +89,47 @@ extension Application {
...
+                if let query = request.url.query {
+                    url += "?\(query)"
+                }
+                var clientRequest = try HTTPClient.Request(
+                    url: url,
+                    method: request.method,
+                    headers: request.headers
+                )
+                clientRequest.body = .byteBuffer(request.body)
+                let response = try await client.execute(request: clientRequest).get()
```
Comment (gwynne) [inline Sources/XCTVapor/XCTApplication.swift:120]: Why is `.get()` needed here? `HTTPClient` has an async version of `execute()` (with different argument labels).

**Review by gwynne — CHANGES_REQUESTED** [review]
One nit

File: Sources/XCTVapor/XCTApplication.swift (hunk 1 above)
Comment (0xTim) [reply Sources/XCTVapor/XCTApplication.swift]: Done

**Comment by 0xTim** [comment]
@gwynne this is ready for you now I think

**Review by gwynne — APPROVED** [review]

**Pull request merged**